src/mem3/src/mem3_reshard.erl

```diff
@@ -438,7 +438,7 @@ reload_job(#job{job_state = JS} = Job, #state{state = stopped} = State) when
     State;
 % This is a case when a job process should be spawend
 reload_job(#job{job_state = JS} = Job, #state{state = running} = State) when
-    JS =:= running orelse JS =:= new
+    JS =:= failed orelse JS =:= running orelse JS =:= new
```

Comment:
If it's failed we wouldn't want to start, we should try to treat it as a "stopped" (i.e. paused) job.

Reply:
like so?

Reply:
yup

It would be good to have a test for it, I was thinking something like https://github.com/apache/couchdb/blob/main/src/mem3/test/eunit/mem3_reshard_api_test.erl#L380-L418

I'll try to make a PR for the PR :-) with a test

Reply:
feel free to commit on this branch

Reply:
added [feat: add restarting failed jobs test](https://github.com/apache/couchdb/pull/4680/commits/a7e09db1930dbd6c82b3cd8764d223106e2f6492)

Reply:
I see a test failure in
```
    mem3_reshard_api_test:643: completed_job_handling...*failed*
in function mem3_reshard_api_test:'-completed_job_handling/1-fun-11-'/2 (test/eunit/mem3_reshard_api_test.erl, line 685)
in call from eunit_test:run_testfun/1 (eunit_test.erl, line 71)
in call from eunit_proc:run_test/1 (eunit_proc.erl, line 531)
in call from eunit_proc:with_timeout/3 (eunit_proc.erl, line 356)
in call from eunit_proc:handle_test/2 (eunit_proc.erl, line 514)
in call from eunit_proc:tests_inorder/3 (eunit_proc.erl, line 456)
in call from eunit_proc:with_timeout/3 (eunit_proc.erl, line 346)
in call from eunit_proc:run_group/2 (eunit_proc.erl, line 570)
**error:{badmatch,{500,
           #{<<"error">> => <<"case_clause">>,
             <<"reason">> =>
                 <<"[{badrpc,\n     {'EXIT',\n         {{{case_clause,complete"...>>,
             <<"ref">> => 3529533974}}}
  output:<<"">>
```